src/ConstructiveSolidGeometry/SurfacePrimitives/ConalPlane.jl

```diff
@@ -41,6 +42,20 @@ function sample(c::ConalPlane{T}, step::Quantity{<:Real, Unitful.𝐋}) where {T
     samples
 end
 
+function sample(c::ConalPlane{T}, Nsamps::NTuple{3,Int}) where {T}
+    samples = CylindricalPoint{T}[]
+    rbotMin::T, rbotMax::T, rtopMin::T, rtopMax::T = get_r_limits(c)
+    zMin::T, zMax::T = get_z_limits(c)
+    #step = T(ustrip(uconvert(u"m", step)))
+    for z in (Nsamps[3] ≤ 1 ? zMin : range(zMin, zMax, length = Nsamps[3]))
+        r_at_z = get_r_at_z(c, z)
+        for r in (Nsamps[3] ≤ 1 ? _left_radial_interval(r_at_z) : range(_left_radial_interval(r_at_z), _right_radial_interval(r_at_z), length = Nsamps[1]))
+            push!(samples, CylindricalPoint{T}(r,c.φ,z))
```

Comment:
Do not use `push!`.
Calculate the number of sample points and allocate the vector `samples` with corresponding length.

Reply:
Partially correct.
We will only only use the sample method with a low number of points to add grid points at the beginning.
Afterwards, we will use the sample (surface) method with a higher number of grid points (depending on the current grid density) to paint the contacts on the grid. 
Because the primitives might get translated and rotated, keep that in mind.
At least, that's how I imagine this right now. We will have to see how things develop. 

That's why I just wanted a simple sample method for the beginning so we can continue and actually find out what will work preferably.

Reply:
So, your proposal is to have a basic `sample` function based on `NTuple{3,Int}` for the moment and write convenience functions for what might come?

Reply:
Well, I would actually add a method which get `ranges/Abstractvector`  over which the loops run to reduce code duplications.
Both current methods `NTuple` and `step` should than calculate the ranges and call the low-level method. 
See comment above :sweat_smile:

Reply:
But same problem: the `r` range will be different at different `z`, for the `Cone{T}` at least.

Reply:
It's a linear scaling, one could just add a scaling parameter to the method.
That was my hole point from the beginning! Just start with the simple `NTuple` version.
Thanks to Julia we can add very easily more complex and advanced sampling methods without having
to change other parts of the code. But, this we should do once we see certain issues with the simple `NTuple` version.

Reply:
Exactly, this happens sometimes. However I could write general 1D sampling methods that we could use wherever possible eg. `sample_z(c::AbstractPrimitive{T}, Nsamps::NTuple{3,Int})`, `sample_z(c::AbstractPrimitive{T}, step::Real)`. These would go in Sampling.jl and call on the eg. `get_z_limits` that we have already written when applicable. We would have to use these with care because something like a `TorusMantle` would not have the `get_z_limits`. Therefore these 1D sampling methods would fail when applied to the incorrect primitive. However, they would be general and only internal and used as much as possible to reduce a lot of code duplications of `get_coordinate_limits`.

Reply:
> It's a linear scaling, one could just add a scaling parameter to the method.
It would need a linear scaling and offset, which I think is overkill to add in the method, we can scale and offset the output when needed. However

> That was my hole point from the beginning! Just start with the simple `NTuple` version.
> Thanks to Julia we can add `later` very easily more complex and advanced sampling methods without having
> to change other parts of the code. But, this we should do once we see certain issues with the simple `NTuple` version.

The step method was the first one I made, so all I did was not delete it.

Reply:
@lmh91 should I write the 1D sampling methods to create ranges as you suggested?

Reply:
No, but keep it in mind as we might want to add it later. 
I want to keep it as simple as possible (this includes as little as code as possible). 
Once we face issues, we can add more advanced solutions. 

The sampling methods you already have you can leave inside. Just please remove the usage of `push!`. 
And make sure to return the correct point types. (@fhagemann comments above regarding the file `sampling.jl`)